pkg/resource/broker/sdk.go

```diff
@@ -565,14 +581,18 @@ func (rm *resourceManager) sdkUpdate(
 		return nil, err
 	}
 
-	resp, respErr := rm.sdkapi.UpdateBrokerWithContext(ctx, input)
-	rm.metrics.RecordAPICall("UPDATE", "UpdateBroker", respErr)
-	if respErr != nil {
-		return nil, respErr
+	var resp *svcsdkapi.UpdateBrokerResponse
+	resp, err = rm.sdkapi.UpdateBrokerWithContext(ctx, input)
+	rm.metrics.RecordAPICall("UPDATE", "UpdateBroker", err)
+	if err != nil {
+		return nil, err
 	}
 	// Merge in the information we read from the API call above to the copy of
 	// the original Kubernetes object we passed to the function
 	ko := desired.ko.DeepCopy()
+	// Copy status from latest observed state
+	latestKOStatus := latest.ko.DeepCopy().Status
+	ko.Status = latestKOStatus
```

Comment:
@A-Hilaly (original) lines 552-554:

```go
	if brokerCreateInProgress(latest) {
		msg := "Broker state is CREATION_IN_PROGRESS"
		setSyncedCondition(desired, corev1.ConditionFalse, &msg, nil)
		return desired, requeueWaitWhileCreating
	}
```

That code means this code will never run if the status is `CREATION_IN_PROGRESS`, no?

Reply:
Yes exactly, the "status copy" code will only get executed once the broker is created (when `brokerState` is `RUNNING`)